contrib/storage-mongo/src/test/java/org/apache/drill/exec/store/mongo/TestTableGenerator.java

```diff
@@ -58,7 +59,16 @@ public static void generateTable(String dbName, String collection,
         .jsonArray(jsonArray).importFile(jsonFile).build();
     MongoImportExecutable importExecutable = MongoImportStarter
         .getDefaultInstance().prepare(mongoImportConfig);
-    importExecutable.start();
+    MongoImportProcess importProcess = importExecutable.start();
+
+    try {
+      while (importProcess.isProcessRunning()) {
+        Thread.sleep(1000);
+      }
+    }catch (Exception ex) {
+      logger.error("Import mongoDb failed", ex);
```

Comment:
Paul, since there are two JIRAs here, I filed a new pull request  https://github.com/apache/drill/pull/741 which addressed your comments. Could you look at that pull request? Thanks.